### PR TITLE
add `**` to jsffi

### PR DESCRIPTION
**Pull request by ringabout**
ref: #12455 and https://github.com/timotheecour/Nim/issues/74

It's better to use jscore though.
```nim
import jscore

echo Math.pow(2, 4) 
# 16.0
```

### CODE REVIEW

**Comment by juancarlospaco** [comment]
This is still needed!, I tried to add them in the past but got rejected because reasons, I hope this get merged soon. :+1:

**Comment by Araq** [comment]
Add a changelog entry.

File: lib/js/jsffi.nim [hunk 1]
```diff
@@ -180,6 +180,7 @@ proc `>`  *(x, y: JsObject): JsObject {.importcpp: "(# > #)".}
 proc `<`  *(x, y: JsObject): JsObject {.importcpp: "(# < #)".}
 proc `>=` *(x, y: JsObject): JsObject {.importcpp: "(# >= #)".}
 proc `<=` *(x, y: JsObject): JsObject {.importcpp: "(# <= #)".}
+proc `**` *(x, y: JsObject): JsObject {.importcpp: "((#) ** #)".}
```
Comment (timotheecour) [inline lib/js/jsffi.nim:183]: pre-existing but why are all these JsObject?
this makes it much less useful.
instead this seems better:
```nim
when true:
  import jsffi
  proc `**`*[T: SomeNumber](x, y: T): T {.importcpp: "((#) ** #)".}
  doAssert 2**3 == 8
  doAssert 2.0**3 == 8.0
```
can be dealt with as a followup PR though
(possibly T: SomeNumber|JsObject altough I don't see the point of JsObject here)

Comment (Clyybber) [reply lib/js/jsffi.nim:183]: That would be worse, Nim doesn't use `**` as the exponentiation operator.

Comment (timotheecour) [reply lib/js/jsffi.nim:183]: here's the original PR form @juancarlospaco : https://github.com/nim-lang/Nim/pull/12455

my comment remains; as @andreaferretti said here:
https://github.com/nim-lang/Nim/pull/12455#discussion_r347258066
> I don't understand this feedback. Exponentiation does not work on JsObject. In javascript you do not exponentiate two objects, you exponentiate two numbers.

to make it more precise, `**` operates on `BigInt | Number`, so in nim side (pending BigInt)  that would be `SomeNumber`

**Pull request merged**